javascript/src/wombatSetup.js

```diff
@@ -0,0 +1,206 @@
+import { fuzzyRules } from './fuzzyRules.js';
+import URI from 'uri-js';
+
+export function applyFuzzyRules(path) {
+  // Apply fuzzy rules to simplify the ZIM path. First matching rule is applied and
+  // result is immediately returned
+
+  for (const rule of fuzzyRules) {
+    const new_path = path.replace(new RegExp(rule.match), rule.replace);
+    if (new_path != path) {
+      return new_path;
+    }
+  }
+  return path;
+}
+
+export function urlRewriteFunction(
+  current_url, // The current (real) url we are on
+  orig_host, // The host of the original url
+  orig_scheme, // The scheme of the original url
+  orig_url, // The original url
+  prefix, // The (absolute) prefix to add to all our urls (from where we are served))
+  url, // first argument passed by wombat.JS at each invocation
+  useRel,
+  mod,
+  doc, // last argument passed by wombat.JS at each invocation
+) {
+  if (!url) return url;
+
+  // Special stuff which is not really a URI but exists in the wild
+  if (['#', '{', '*'].includes(url.substring(0, 1))) return url;
+
+  // If URI scheme is defined but not http or https, we have to not rewrite the URL
+  const uri = URI.parse(url);
+  if (
+    typeof uri.scheme !== 'undefined' &&
+    !['http', 'https'].includes(uri.scheme)
+  )
+    return url;
+
+  // If url starts with prefix, we need to remove this prefix before applying usual
+  // rewrite rules
+  if (url.startsWith(prefix)) {
+    url = uri.scheme + '://' + url.substring(prefix.length);
```

Comment:
Can we mention above (in prefix comment) what prefix looks like.

Reply:
I've added sample value for all args indeed ^^